### DISCUSSION TITLE
Support `pages/_document` as a function component

### DISCUSSION THREAD

**Discussion by timneutkens** [vercel/next.js#26423] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

Converting `_document` to a function component has no real benefits currently besides being able to use `useContext` but there's no context wrapper above _document that you can use with useContext. `useState`, `useEffect`, and other hooks will not work as well given that `_document` is not hydrated client-side. 

Even though there are no benefits the question to support a function instead of a class comes up from time to time in #7515, hence why I'm creating this story.

## Describe the solution you'd like

`_document` is currently not a function component for backwards compat reasons, we can likely take a similar approach to what we did for _app though by having getInitialProps fall back to the default value.

**Comment by hasanzahran** [comment]
Is there any update regarding this issue.

Thanks

**Comment by kimgiutae** [comment]
Is this realated to this recent PR? https://github.com/vercel/next.js/pull/28515

**Reply by FunctionDJ** [reply]
Seems like it. It works for me - no class component in my source anymore.
